scripts/vagrant/scripts/configureK8smaster.sh

```diff
@@ -1,15 +1,27 @@
 #!/bin/bash
-# Get the IP address that VirtualBox has given this VM
-IPADDR=$(ifconfig eth1 | grep -i Mask | awk '{print $2}'| cut -f2 -d:)
-echo This VM has IP address "$IPADDR"
+
+# Set whether or not to use IPv6 enabled Kubernetes deployment
+ENABLE_IPV6=0
+POD_CIDR="10.32.0.0/12"
 
 # Setup Hugepages
 #echo "Copying /vagrant/10-kubeadm.conf to /etc/systemd/system/kubelet.service.d/10-kubeadm.conf"
 #cp /vagrant/10-kubeadm.conf /etc/systemd/system/kubelet.service.d/10-kubeadm.conf
 
 # Set up Kubernetes
 NODENAME=$(hostname -s)
-kubeadm init --apiserver-cert-extra-sans="$IPADDR" --apiserver-advertise-address="$IPADDR" --node-name "$NODENAME" --pod-network-cidr="10.32.0.0/12"
+
+# Get the IP address that VirtualBox has given this VM
+if [ "$ENABLE_IPV6" -eq 1 ]; then
+    echo "Deploying Kubernetes with IPv6..."
+    IPADDR=$(ip -6 addr|awk '{print $2}'|grep -P '^(?!fe80)[[:alnum:]]{4}:.*/64'|cut -d '/' -f1)
+    sysctl -w net.ipv6.conf.all.forwarding=1
+else
+    IPADDR=$(ifconfig eth1 | grep -i Mask | awk '{print $2}'| cut -f2 -d:)
+fi
+echo This VM has IP address "$IPADDR"
+
+kubeadm init --apiserver-cert-extra-sans="$IPADDR" --apiserver-advertise-address="$IPADDR" --node-name "$NODENAME" --pod-network-cidr="$POD_CIDR"
```

Comment:
You currently propose to have an IPv4 `POD_CIDR` with the option to get IPv6 API server. I wonder if `nsmd-k8s` will be able to talk to the API server.